src/policy/protect_payload.rs

```diff
@@ -212,3 +216,87 @@ impl ProtectPayloadPolicy {
         }
     }
 }
+
+// ———————————————————————————————— Explicit Forwarding Rules ———————————————————————————————— //
+
+#[derive(Clone)]
+pub struct ForwardingRule {
+    mcause: MCause,
+    allow_in: [bool; 32],
+    allow_out: [bool; 32],
+}
+
+impl ForwardingRule {
+    pub const NB_RULES: usize = 1;
+
+    fn match_rule(trap_cause: MCause, rules: &mut [ForwardingRule; 1]) -> ForwardingRule {
+        for rule in rules {
+            if trap_cause == rule.mcause {
+                return rule.clone();
+            }
+        }
+
+        Self::new_allow_nothing(trap_cause)
+    }
+
+    fn build_forwarding_rules() -> [ForwardingRule; Self::NB_RULES] {
+        let mut rules = [Self::new_allow_nothing(EcallFromSMode); 1];
+
+        // Build Ecall rule
+        rules[0]
+            .allow_register_in(Register::X10)
+            .allow_register_in(Register::X11)
+            .allow_register_in(Register::X12)
+            .allow_register_in(Register::X13)
+            .allow_register_in(Register::X14)
+            .allow_register_in(Register::X15)
+            .allow_register_in(Register::X16)
+            .allow_register_in(Register::X17)
+            .allow_register_out(Register::X10)
+            .allow_register_out(Register::X11);
+
+        rules
+    }
+
+    #[allow(unused)]
+    fn new_allow_nothing(mcause: MCause) -> Self {
+        ForwardingRule {
+            mcause,
+            allow_in: [false; 32],
+            allow_out: [false; 32],
+        }
+    }
+
+    #[allow(unused)]
+    fn new_allow_everything(mcause: MCause) -> Self {
+        ForwardingRule {
+            mcause,
+            allow_in: [true; 32],
+            allow_out: [true; 32],
+        }
+    }
+
+    #[allow(unused)]
+    fn allow_register_in(&mut self, reg: Register) -> &mut Self {
+        self.allow_in[reg as usize] = true;
+        self
+    }
+
+    #[allow(unused)]
+    fn block_register_in(&mut self, reg: Register) -> &mut Self {
+        self.allow_in[reg as usize] = false;
+        self
+    }
+
+    #[allow(unused)]
+    fn allow_register_out(&mut self, reg: Register) -> &mut Self {
+        self.allow_out[reg as usize] = true;
+        self
+    }
+
+    #[allow(unused)]
+    fn block_register_out(&mut self, reg: Register) -> &mut Self {
+        self.allow_out[reg as usize] = false;
+        self
+    }
```

Comment:
In general, let's remove the `#[allow]` annotation and only keep the one we actually use.

The problem with those `#[allow]` annotation is that we always end-up forgetting to remove then, and then we end-up with a lot of dead-code that the compile doesn't warn about...